src/main/kotlin/com/hibob/academy/dao/PetDao.kt

```diff
@@ -0,0 +1,45 @@
+package com.hibob.academy.dao
+
+import org.jooq.DSLContext
+import org.springframework.stereotype.Component
+import jakarta.inject.Inject
+import org.jooq.Record
+import org.jooq.RecordMapper
+import java.time.LocalDate
+
+// todo dao create update + tests pr 1 + test the fetch from yesterday)
+//
+
+@Component
+class PetDao @Inject constructor(private val sql: DSLContext) {
+
+    private val petTable = PetTable.petInstance
+
+    private val petTableMapper = RecordMapper<Record, Pet> { record ->
+        Pet(
+            id = record[petTable.id],
+            name = record[petTable.name],
+            type = record[petTable.type],
+            companyId = record[petTable.companyId],
+            dateOfArrival = record[petTable.dateOfArrival]
+        )
+    }
+
+    fun createPet(name: String, type: String, companyId: Long, dateOfArrival: LocalDate) {
```

Comment:
Let keep use PetType instead of string and convert it to string in the function.

Reply:
Missed it! thanks